client/src/partials/patients/groups/groups.js

```diff
@@ -1,127 +1,126 @@
-    };
-
-    appstate.register('enterprise', loadEnterprise);
-
-    function loadEnterprise(enterprise) {
-      $scope.enterprise = enterprise;
-      dependencies.group.query.where = ['patient_group.enterprise_id='+enterprise.id];
-      validate.process(dependencies).then(initialisePatientGroup);
-    }
+.controller('patientGroupController', patientGroupController);
 
-    function initialisePatientGroup(model) {
-      angular.extend($scope, model);
-    }
+/** inject service(s) or dependencies for the modules **/
+patientGroupController.$inject = ['PatientGroupService', 'PriceListService', 'SessionService', '$window'];
 
-    $scope.remove = function (grp) {
-      if (!$window.confirm($translate.instant('PATIENT_GRP.CONFIRM_MESSAGE'))) {
-        return;
-      }
+function patientGroupController (patientGroupService, priceListService, sessionService, $window){
+  var vm = this; 
 
-      connect.delete('patient_group', 'uuid', [grp.uuid])
-      .then(function () {
-        $scope.group.remove(grp.uuid);
-      }, function (err) {
-        messenger.danger('error:' + JSON.stringify(err));
-      });
-    };
-
-    // register namespace
-
-    $scope.newGroup = function () {
-      $scope.register = {};
-      $scope.action = 'register';
-      session.selected = null;
-    };
-
-    $scope.saveRegistration = function () {
-      var packaged = connect.clean($scope.register);
-      packaged.uuid = uuid();
-      packaged.enterprise_id = $scope.enterprise.id;
-
-      // validate that register is complete.
-      connect.post('patient_group', packaged)
-      .then(function () {
-        $scope.group.post(packaged);
-        $scope.edit(packaged);
-      });
-    };
+  /** This method is responsible of initializing data **/
+  function startup (){
+
+    /** make the loading state into true, while loading data **/
+    vm.loading = true;
+
+    /** fetching all price list **/
+    priceListService.read()
+    .then(function (priceLists){
+
+      /** attaching the price list to the view **/
+      vm.priceLists = priceLists;
 
-    $scope.resetRegistration = function () {
-      $scope.register = {};
-    };
+      /** load all patient groups**/
+      return loadPatientGroup();
+    })
+    .then(function (patientGroups){     
+
+      vm.groups = patientGroups;
+
+      /** turn loading to false, we all data **/
+      vm.loading = false;
+    });
+  }
+
+  /** this method is responsible to propose a GUI to user for creation **/
+  function create (){
+
+    /** init the patient group **/
+    vm.patientGroup = {};
+
+    /** swicth the view to create **/    
+    vm.view = 'create';
+  }
 
-    // edit namespace
+  /** this function is responsible of submiting the patient group to the server for creation **/
+  function submit (invalid){
 
-    $scope.edit = function (grp) {
-      $scope.action = 'modify';
-      $scope.modify = angular.copy(grp);
-      $scope.modify_original = grp;
+    /** if the form is not valid do nothing **/
```

Comment:
This comment really helps with code readability :+1: 

I've noticed this pattern in a number of places [https://github.com/IMA-WorldHealth/bhima-2.X/pull/196#discussion_r57133716](here) and [https://github.com/IMA-WorldHealth/bhima-2.X/pull/200#discussion_r57083356](here) .

If those small changes could be applied it should read clearly.